### PR TITLE
Add rt1180 hyperram support

### PR DESCRIPTION
**Pull request by lucien-nxp**
Support use xmcd to initialize the external hyperram on board.

### CODE REVIEW

File: dts/arm/nxp/nxp_rt118x.dtsi [hunk 1]
```diff
@@ -100,3 +220,29 @@
...
+
+&flexspi2 {
+	compatible = "nxp,imx-flexspi";
+	interrupts = <56 0>;
+	#address-cells = <1>;
+	#size-cells = <0>;
+	status = "disabled";
+	clocks = <&ccm IMX_CCM_FLEXSPI2_CLK 0x0 0>;
+};
+
```
Comment (dleach02) [inline dts/arm/nxp/nxp_rt118x.dtsi:232]: I don't think the changes above this point are related to the PR objective?

File: dts/arm/nxp/nxp_rt118x_cm33.dtsi [hunk 2]
```diff
@@ -27,6 +33,10 @@
 		flexspi1: spi@525e0000 {
 			reg = <0x525e0000 0x4000>,<0x38000000 DT_SIZE_M(128)>;
 		};
+
```
Comment (dleach02) [inline dts/arm/nxp/nxp_rt118x_cm33.dtsi:36]: There are changes in this commit and file that do not appear to be related to the the hyperram support.

File: dts/arm/nxp/nxp_rt118x_cm33_ns.dtsi [hunk 3]
```diff
@@ -80,6 +89,75 @@
 		<&iomuxc_aon_gpio_aon_27_gpio1_io27>;
 };
 
+&gpio2{
```
Comment (dleach02) [inline dts/arm/nxp/nxp_rt118x_cm33_ns.dtsi:92]: ditto

File: soc/nxp/imxrt/CMakeLists.txt [hunk 4]
```diff
@@ -53,6 +63,7 @@ if(CONFIG_SOC_SERIES_IMXRT118X)
     PRIO 11
   )
   zephyr_compile_definitions(XIP_EXTERNAL_FLASH)
+  zephyr_compile_definitions(USE_HYPERRAM)
```
Comment (dleach02) [inline soc/nxp/imxrt/CMakeLists.txt:66]: should this be driven by a CONFIG?

File: soc/nxp/imxrt/imxrt118x/linker.ld [hunk 5]
```diff
@@ -12,8 +12,8 @@
 MEMORY
      {
 
-#if (DT_REG_SIZE(DT_NODELABEL(sdram0)) > 0) && !IS_CHOSEN_SRAM(sdram0)
```
Comment (dleach02) [inline soc/nxp/imxrt/imxrt118x/linker.ld:15]: I think the soc files should be flexible and driven by DT & kconfig to give folks flexibility at the application level to use what they want.

File: dts/arm/nxp/nxp_rt118x.dtsi (hunk 1 above)
Comment (lucien-nxp) [reply dts/arm/nxp/nxp_rt118x.dtsi:232]: Sorry, originally thought that together can save a PR review process.  I have created another PR to complement full LPUART and GPIO devices, and ocram spaces. 
https://github.com/zephyrproject-rtos/zephyr/pull/77258

File: soc/nxp/imxrt/imxrt118x/linker.ld (hunk 5 above)
Comment (lucien-nxp) [reply soc/nxp/imxrt/imxrt118x/linker.ld:15]: SDRAM on RT1180 isn't out of box. So I think we don't need to consider SDRAM on RT1180 EVK. Default connection on boards is hyperram, so I change the linker description for hyperram, if user don't use hyperram as default RAM region, hyperram region will still display in linker memory region.

Comment (danieldegrasse) [reply soc/nxp/imxrt/imxrt118x/linker.ld:15]: Sure, but this is an SOC support file. Since the SOC can support SDRAM, shouldn't we keep the definition in the linker file? It seems like all this will do is add an SDRAM linker region if SDRAM is declared in the board devicetree with a nonzero size. Of course, the linker region could also be handled automatically if we use the `zephyr,memory-region` compatible- so this might not be necessary at all.

Comment (lucien-nxp) [reply soc/nxp/imxrt/imxrt118x/linker.ld]: Agreed. I have added back the sdram region (if exist) description in linker file. This ensures that the SOC folder is more static in the future when customers may need support for SDRAM.

File: soc/nxp/imxrt/CMakeLists.txt (hunk 4 above)
Comment (danieldegrasse) [reply soc/nxp/imxrt/CMakeLists.txt:67]: The `zephyr_compile_definitions(USE_HYPERRAM)` should be set at the board level, not SOC level. This C compiler definition is being used by our board specific file with XCMD setup commands to configure hyperram. From what I can tell, no file generic to the SOC depend on `USE_HYPERRAM` being defined, right? So a user could enable hyperram on a custom board without defining `USE_HYPERRAM`?

Comment (lucien-nxp) [reply soc/nxp/imxrt/CMakeLists.txt]: Move `zephyr_compile_definitions(USE_HYPERRAM)` defined to board level.
Yes, `USE_HYPERRAM` is needed in SDK board driver if want to use hyperram.

**Review by danieldegrasse — APPROVED** [review]

**Review by dleach02 — APPROVED** [review]
